packages/big-design/src/components/Table/spec.tsx

```diff
@@ -197,90 +197,219 @@ test('renders the total number of items + item name', () => {
-  expect(nextButtonPage).toBeVisible();
+    expect(onNext).toHaveBeenCalled();
+  });
+
+  test('renders a pagination component with custom button labels', async () => {
+    render(
+      <Table
+        columns={[
+          { header: 'Sku', hash: 'sku', render: ({ sku }) => sku },
+          { header: 'Name', hash: 'name', render: ({ name }) => name },
+          { header: 'Stock', hash: 'stock', render: ({ stock }) => stock },
+        ]}
+        items={[
+          { sku: 'SM13', name: '[Sample] Smith Journal 13', stock: 25 },
+          { sku: 'DPB', name: '[Sample] Dustpan & Brush', stock: 34 },
+          { sku: 'OFSUC', name: '[Sample] Utility Caddy', stock: 45 },
+          { sku: 'CLC', name: '[Sample] Canvas Laundry Cart', stock: 2 },
+          { sku: 'CGLD', name: '[Sample] Laundry Detergent', stock: 29 },
+        ]}
+        pagination={{
+          currentPage: 1,
+          itemsPerPage: 3,
+          itemsPerPageOptions: [3, 5, 10],
+          onItemsPerPageChange: jest.fn(),
+          onNext: jest.fn(),
+          onPrevious: jest.fn(),
+          localization: {
+            label: '[Custom] Pagination',
+            previousPage: '[Custom] Previous page',
+            nextPage: '[Custom] Next page',
+            rangeLabel: '[Custom] Change items per page',
+          },
+        }}
+      />,
+    );
+
+    const navigation = await screen.findByRole('navigation', { name: '[Custom] Pagination' });
+    const paginationDropdown = await screen.findByRole('button', {
+      name: '[Custom] Change items per page',
+    });
+    const previousButtonPage = await screen.findByRole('button', {
+      name: '[Custom] Previous page',
+    });
+    const nextButtonPage = await screen.findByRole('button', { name: '[Custom] Next page' });
+
+    expect(navigation).toBeVisible();
+    expect(paginationDropdown).toBeVisible();
+    expect(previousButtonPage).toBeVisible();
+    expect(nextButtonPage).toBeVisible();
+  });
+
+  describe('selectable', () => {
+    test('renders a summary of the selected items, without reference to the total number of items', () => {
+      const selectedItems = [
+        { sku: 'CLC', name: '[Sample] Canvas Laundry Cart', stock: 2 },
+        { sku: 'CGLD', name: '[Sample] Laundry Detergent', stock: 29 },
+      ];
+
+      render(
+        <Table
+          columns={[
+            { header: 'Sku', hash: 'sku', render: ({ sku }) => sku },
+            { header: 'Name', hash: 'name', render: ({ name }) => name },
+            { header: 'Stock', hash: 'stock', render: ({ stock }) => stock },
+          ]}
+          itemName="Product"
+          items={[
+            { sku: 'SM13', name: '[Sample] Smith Journal 13', stock: 25 },
+            { sku: 'DPB', name: '[Sample] Dustpan & Brush', stock: 34 },
+            { sku: 'OFSUC', name: '[Sample] Utility Caddy', stock: 45 },
+            { sku: 'CLC', name: '[Sample] Canvas Laundry Cart', stock: 2 },
+            ...selectedItems,
+          ]}
```

Comment:
Just to clarify, what does providing an overlapping product between `items` and `selectedItems` prove about the table's behaviour?

Reply:
it's more a requirement of `Table`'s implementation

inside it does an strict equality check, so they literally need to be the same objects to class as 'selected'

Reply:
Ah, yes, the good old internal `Set`...